`resource.struct_rusage` stub is named `resource._RUsage` for some reason

The first occurrence of the name `resoruce._RUsage` that I was able to find was commited into `python/mypy` stubs on [21 Aug 2015](https://github.com/python/mypy/commit/6282eb0a04dc254c3531a1dbe482b8633dad09ec#diff-de44ba7f37f2aedfcdd858a825291f635d980726e807a278e66ca6d63070e880R22). It was later copied over to this repo on [30 Sep 2015](https://github.com/python/typeshed/commit/337abed05a35f0c5b67c02d42bca0af1e2adc76e#diff-a5943a811cac3726f5cb299498819dd9412dbb6d3ef392444c690b4f2861a426R22) and then copied to the Python 3 stubs on [22 Aug 2017](https://github.com/python/typeshed/commit/485c232409533982369e5d56493540d00519a2b9#diff-5a0da7d8759b759be9d2b61c2ef947cf8a246376492727e75d66f56a6ce9edf6R28).

Type `resource.struct_rusage` that aformentioned stub describes has been added to the Python source code on [9 Apr 2002](https://github.com/python/cpython/commit/688357e035d9cb291c834d14ac5acc03905b1665#diff-510472560baed5d8506ea0cde42fbb06c286a90dc85ce3e6172cb088de5d5ad8R47), and has never been named `_RUsage`.

Because of this, files that use `resource.struct_rusage` get yelled at by `mypy`:
```
$  cat test2.py
from resource import struct_rusage
$  python test2.py
$  mypy test2.py
test2.py:1: error: Module 'resource' has no attribute 'struct_rusage'
Found 1 error in 1 file (checked 1 source file)
```
And files that use `resource._RUsage` get yelled at by `python`:
```
$  cat test1.py
from resource import _RUsage
$  mypy test1.py
Success: no issues found in 1 source file
$  python test1.py
Traceback (most recent call last):
  File "/tmp/test1.py", line 1, in <module>
    from resource import _RUsage
ImportError: cannot import name '_RUsage' from 'resource' (/usr/lib/python3.9/lib-dynload/resource.cpython-39-x86_64-linux-gnu.so)
```

--- a/stdlib/resource.pyi
+++ b/stdlib/resource.pyi
@@ -1,5 +1,5 @@
 import sys
-from typing import NamedTuple, Tuple, overload
+from typing import Any, Dict, Tuple, overload
 
 RLIMIT_AS: int
 RLIMIT_CORE: int
@@ -23,7 +23,10 @@ if sys.platform == "linux":
     RLIMIT_SIGPENDING: int
     RUSAGE_THREAD: int
 
-class _RUsage(NamedTuple):
+_Tuple16 = Tuple[float, float, int, int, int, int, int, int, int, int, int, int, int, int, int, int]
+
+class struct_rusage(_Tuple16):
+    def __new__(cls, sequence: _Tuple16, dict: Dict[str, Any] = ...) -> struct_rusage: ...
     ru_utime: float
     ru_stime: float
     ru_maxrss: int
@@ -43,7 +46,7 @@ class _RUsage(NamedTuple):
 
 def getpagesize() -> int: ...
 def getrlimit(__resource: int) -> Tuple[int, int]: ...
-def getrusage(__who: int) -> _RUsage: ...
+def getrusage(__who: int) -> struct_rusage: ...
 def setrlimit(__resource: int, __limits: Tuple[int, int]) -> None: ...
 
 if sys.platform == "linux":

